mergify_engine/check_api.py

```diff
@@ -113,22 +114,53 @@ class Result:
     ended_at: typing.Optional[datetime.datetime] = None
 
 
+def to_check_run_light(
```

Comment:
I don't know how hard it is, but in theory calling `del` on the original and then casting it would be more efficient.

Reply:
Using del on typing.TypedDict is mypy hell

Reply:
you can't build a TypeDict from another TypedDict, the only ways is keeping typing.Any, delete keys, typing.cast. And you are not sure about what the dict really contains it you missing to del some keys or GitHub add new keys.

Reply:
So copying is a dumb from memory handling perspective, but at least the final dict is the expected for sure.